content/articles/python-decorators/index.md

```diff
@@ -0,0 +1,295 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/python-decorators/
+title: Getting started with decorators in Python
+description: This tutorial will be a brief dive into the concept of python decorators. how to use them and their advantages to programmers.
+author: mia-roberts
+date: 2021-08-10T00:00:00-13:30
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/python-decorators/hero.jpg
+    alt: Python decorators image
+---
+Python decorators are a design pattern that allows the developer to modify the structure of a function without changing the function's code structure. 
+<!--more-->
+This concept is useful when you want to change the behavior of a function, so rather than going in and changing the entire cod, you create a python decorator which will allow you to use a single line of code to change the behavior of all different functions.
+
+### Prerequisites
+Appreciating that everything in Python programming comes as an object is a key to understanding this concept. The reader should be well conversant with Python, although the explanations are simple and precise for beginners.
+
+Besides that, you need:
+- A working installation of Python on your computer
+- A code editor to try out the snippets
+
+### Understanding what decorators
+The concept of Python decorators is unclear to many people, even though it's simple. For this reason, you can consider Python decorators as high-level functions in Python which that takes another function as its argument and returns another function as a return value.
```

Comment:
Python dcorators being unclear is not a reason to make it a high level function.  Give a good reason as to why moajority of programmers do not understand python decorators.